framework/SupervisedLearning/SupervisedLearning.py

```diff
@@ -131,6 +131,7 @@ def __init__(self):
     self.muAndSigmaFeatures = {}   # normalization parameters
     self.metadataKeys = set()      # keys that can be passed to DataObject as meta information
     self.metadataParams = {}       # indexMap for metadataKeys to pass to a DataObject as meta dimensionality
+    self.needsDictTraining = True  # True if the "train" method expects a dictionary ONLY
```

Comment:
Here's the defaulting declaration of the legacy node flag. Existing SVLs will all use dicts to train, but this should be discouraged for future algorithms.